checkov/common/output/cyclonedx.py

```diff
@@ -91,9 +60,27 @@ def create_bom(self) -> Bom:
         bom.metadata.tools.add(this_tool)
 
         for report in self.reports:
-            if report.check_type == CheckType.SCA_PACKAGE and self.export_iac_only:
+            if report.check_type in SCA_CHECKTYPES and self.export_iac_only:
                 continue
 
+            if report.check_type == CheckType.SCA_IMAGE:
+                image_record = next(itertools.chain(report.failed_checks, report.passed_checks, report.skipped_checks))
+                image_id = cast(Dict[str, Any], image_record.vulnerability_details).get('image_details', ImageDetails).image_id
+                file_path = image_record.file_path.split(' ')[0]
+                image_purl = PackageURL(
+                    type='oci',
+                    namespace=self.repo_id,
+                    name=file_path,
+                    version=image_id
+                )
+                bom.metadata.component = Component(
```

Comment:
shouldn't the component be added like this `bom.components.add(component)`

Reply:
No, because this is a metadata component. The metadata contains the images themsevles as components, while the regular BOM components are the packages ones with vulnerabilities.
![image](https://user-images.githubusercontent.com/63583491/182577986-00be5fbf-f5fb-477d-8084-391f20c99571.png)

Reply:
gotcha 👍  but won't work for us, since we can have several images in one report